File: src/ChangeMe.Backend/src/ChangeMe.Backend.Domain/Authorization/PermissionCodes.cs
namespace ChangeMe.Backend.Domain.Authorization;

public static class PermissionCodes
{
  public const string UsersView = "Users.View";
  public const string UsersManage = "Users.Manage";
  public const string UsersDeactivate = "Users.Deactivate";
  public const string UsersInvite = "Users.Invite";
  public const string RolesView = "Roles.View";
  public const string RolesManage = "Roles.Manage";
  public const string SessionsViewOwn = "Sessions.ViewOwn";
  public const string SessionsManageOwn = "Sessions.ManageOwn";
  public const string SessionsViewAny = "Sessions.ViewAny";
  public const string SessionsManageAny = "Sessions.ManageAny";
  public const string IssuesView = "Issues.View";
  public const string IssuesCreate = "Issues.Create";
  public const string IssuesEdit = "Issues.Edit";
  public const string IssuesDelete = "Issues.Delete";
  public const string IssuesComment = "Issues.Comment";
  public const string IssuesManageAttachments = "Issues.ManageAttachments";

  public static readonly IReadOnlyList<string> All =
  [
    UsersView,
    UsersManage,
    UsersDeactivate,
    UsersInvite,
    RolesView,
    RolesManage,
    SessionsViewOwn,
    SessionsManageOwn,
    SessionsViewAny,
    SessionsManageAny,
    IssuesView,
    IssuesCreate,
    IssuesEdit,
    IssuesDelete,
    IssuesComment,
    IssuesManageAttachments
  ];

  public static readonly IReadOnlyList<string> DefaultUserRole =
  [
    SessionsViewOwn,
    SessionsManageOwn,
    IssuesView,
    IssuesCreate,
    IssuesComment
  ];
}

File: src/ChangeMe.Frontend/src/app/shared/authorization/permission-codes.ts
export const PermissionCodes = {
  usersView: 'Users.View',
  usersManage: 'Users.Manage',
  usersDeactivate: 'Users.Deactivate',
  usersInvite: 'Users.Invite',
  rolesView: 'Roles.View',
  rolesManage: 'Roles.Manage',
  sessionsViewOwn: 'Sessions.ViewOwn',
  sessionsManageOwn: 'Sessions.ManageOwn',
  sessionsViewAny: 'Sessions.ViewAny',
  sessionsManageAny: 'Sessions.ManageAny',
  issuesView: 'Issues.View',
  issuesCreate: 'Issues.Create',
  issuesEdit: 'Issues.Edit',
  issuesDelete: 'Issues.Delete',
  issuesComment: 'Issues.Comment',
  issuesManageAttachments: 'Issues.ManageAttachments'
} as const;
